src/components/RadioButton/RadioButton.st.css

```diff
@@ -198,10 +198,26 @@
 
 /* Create the indicator (the dot/circle - hidden when not checked) */
 
+.root:box:hover:checked .wrapper {
+    border-color: applyOpacity(value(CurrentIconColor), 1);
+}
+
+.root:box:hover:not(:checked) .wrapper {
+    border-color: applyOpacity(value(CurrentBorderColor), 1);
+}
```

Comment:
And then maybe we don't need the `:not(:checked)` her.
Maybe

Reply:
without the :not(:checked), the color of the border would change when hovering (gets 'CurrentIconColor' instead of 'CurrentBorderColor')